tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1800,23 +1822,169 @@ public:
    }
 
    /**
-    * @brief Create a new SnapshotHelperMT with a different output file name
+    * @brief Create a new SnapshotTTreeHelperMT with a different output file name
+    *
+    * @param newName A type-erased string with the output file name
+    * @return SnapshotTTreeHelperMT
+    *
+    * This MakeNew implementation is tied to the cloning feature of actions
+    * of the computation graph. In particular, cloning a Snapshot node usually
+    * also involves changing the name of the output file, otherwise the cloned
+    * Snapshot would overwrite the same file.
+    */
+   SnapshotTTreeHelperMT MakeNew(void *newName)
+   {
+      const std::string finalName = *reinterpret_cast<const std::string *>(newName);
+      return SnapshotTTreeHelperMT{fNSlots,           finalName,          fDirName, fTreeName,
+                                   fInputBranchNames, fOutputBranchNames, fOptions, std::vector<bool>(fIsDefine)};
+   }
+};
+
+#ifdef R__HAS_ROOT7
+void ValidateSnapshotRNTupleOutput(const RSnapshotOptions &opts, const std::string &ntupleName,
+                                   const std::string &fileName);
+
+/// Helper function to update the value of an RNTuple's field in the provided entry.
+template <typename T>
+void SetFieldsHelper(T value, std::string_view fieldName, ROOT::Experimental::REntry *entry)
+{
+   entry->BindValue(fieldName, std::make_shared<T>(value));
+}
+
+/// Helper object for a single-thread RNTuple-based Snapshot action
+template <typename... ColTypes>
+class R__CLING_PTRCHECK(off) SnapshotRNTupleHelper : public RActionImpl<SnapshotRNTupleHelper<ColTypes...>> {
+   std::string fFileName;
+   std::string fNTupleName;
+
+   std::unique_ptr<TFile> fOutFile = nullptr;
+
+   RSnapshotOptions fOptions;
+   ROOT::Detail::RDF::RLoopManager *fLoopManager;
+   ColumnNames_t fInputFieldNames; // This contains the resolved aliases
+   ColumnNames_t fOutputFieldNames;
+   std::unique_ptr<ROOT::Experimental::RNTupleWriter> fWriter{nullptr};
+
+   ROOT::Experimental::REntry *fOutputEntry;
+
+   std::vector<bool> fIsDefine;
+
+public:
+   using ColumnTypes_t = TypeList<ColTypes...>;
+   SnapshotRNTupleHelper(std::string_view filename, std::string_view ntuplename, const ColumnNames_t &vfnames,
```

Comment:
Do we intentionally not support storing the snapshot into a sub-directory?

Reply:
It is supported for TTree, probably we should try to have at least some support for RNTuple (tests can be added in a follow-up PR too)

Reply:
The problem is that the current RNTuple interfaces barely support it. `RNTupleWriter::Append` does take TDirectory as its argument, but there is no interface to create an `RPageSource` for an RNTuple that's stored in a subdirectory. So I think this requires more thought outside of this PR. However, to make the user experience a bit friendlier, I will add a check and throw in case one attemps to snapshot to a subdirectory.

Reply:
Where did the throw ended up?

Reply:
I've added it in `RInterface.hxx`:

https://github.com/enirolf/root/blob/bf7088ea7bc4a25d41c36dec2e13280c3cda61e2/tree/dataframe/inc/ROOT/RDF/RInterface.hxx#L1357-L1360

and

https://github.com/enirolf/root/blob/bf7088ea7bc4a25d41c36dec2e13280c3cda61e2/tree/dataframe/inc/ROOT/RDF/RInterface.hxx#L3253-L3256

Reply:
> there is no interface to create an `RPageSource` for an RNTuple that's stored in a subdirectory

Is that still the case? If you have an anchor it will natively work, and `RMiniFileReader::GetNTupleProper` parses forward slashes in the `ntuplePath` to mean that it should look in a sub-directory. Not necessarily required in this PR, just trying to understand what is actually needed.

Reply:
That would be an option indeed, we could incorporate it into `RNTupleDS` first and then (re-)add the support here.